src/facets/ProfitAndLossReporterFacet.sol

```diff
@@ -22,104 +23,35 @@ contract ProfitAndLossReporterFacet is TreasuryStorage, ReentrancyGuardUpgradeab
-    /// @return profit The cumulative profit for the previous epoch
-    function profitLatestEpoch() public view returns (uint256 profit) {
-        TreasuryStorage.TreasuryStorageStruct storage $ = _getStorage();
-        uint256 blocks = block.number - $.sUSX.lastEpochBlock();
-        return profitPerBlock() * blocks;
-    }
-
-    /// @notice Calculates the profit per block for the current epoch, to be added to USX balance over time in sharePrice() function
-    /// @return The profit per block for the current epoch
-    function profitPerBlock() public view returns (uint256) {
-        TreasuryStorage.TreasuryStorageStruct storage $ = _getStorage();
-        uint256 epochEnd = $.sUSX.lastEpochBlock() + $.sUSX.epochDuration();
-        if (block.number >= epochEnd) return 0;
-        uint256 blocksRemainingInEpoch = epochEnd - block.number;
-        return blocksRemainingInEpoch == 0 ? 0 : $.netEpochProfits / blocksRemainingInEpoch;
-    }
-
-    /// @notice Calculates the cumulative profits for previous epoch that needs to be still distributed
-    /// @dev Determines what is the USX balance removed in calculating sharePrice() calculated as profitPerBlock(current_block - lastEpochBlock)
-    /// @return profitToSubtract The cumulative profits for previous epoch that needs to be still distributed in USDC
-    function substractProfitLatestEpoch() public view returns (uint256 profitToSubtract) {
+    /// @notice Calculates the insurance fund for the Insurance Fund based on insuranceFundFraction
+    /// @param profitAmount The amount of profits to calculate the insurance fund for
+    /// @return The insurance fund for the Insurance Fund
+    function insuranceFund(uint256 profitAmount) public view returns (uint256) {
         TreasuryStorage.TreasuryStorageStruct storage $ = _getStorage();
-
-        uint256 finalBlock = $.sUSX.lastEpochBlock() + $.sUSX.epochDuration();
-        uint256 currentBlock = block.number;
-        if (currentBlock >= finalBlock || $.netEpochProfits == 0) {
-            return 0;
-        }
-        uint256 blocks = finalBlock - currentBlock; // should substract all profits at the beginning of a new epoch
-        profitToSubtract = profitPerBlock() * blocks; // all regular math checks required
+        return Math.mulDiv(profitAmount, $.insuranceFundFraction, 1000000, Math.Rounding.Floor);
     }
 
     /*=========================== Asset Manager Functions =========================*/
 
-    /// @notice Asset Manager reports total balance of USDC they hold, profits or losses calculated from this value
-    /// @param totalBalance The total balance of USDC held by the Asset Manager
-    function assetManagerReport(uint256 totalBalance) public onlyAssetManager nonReentrant {
+    /// @notice Asset Manager reports profits or losses
+    /// @param profitLoss The profits or losses of the Asset Manager
+    function assetManagerReport(int256 profitLoss) public onlyReporter nonReentrant {
```

Comment:
```suggestion
    function assetManagerReport(uint256 profit) public onlyReporter nonReentrant {
```

We should change this contract to be simply profit / reward distributor instead of a reporter. So no need to allow negative values